droidlet/interpreter/robot/tasks.py

```diff
@@ -359,6 +365,159 @@ def step(self):
     def __repr__(self):
         return "<TrajectorySaverTask {}>".format(self.target)
 
+class CuriousExplore(TrajectorySaverTask):
+    """use slam to explore environemt, but also examine detections"""
+
+    def __init__(self, agent, task_data):
+        super().__init__(agent, task_data)
+        self.steps = ["not_started"] * 2
+        self.task_data = task_data
+        self.goal = task_data.get("goal", (19,19,0))
+        self.init_curious_logger()
+        self.agent = agent
+        self.objects_examined = 0
+        self.save_data = task_data.get('save_data')
+        print(f'CuriousExplore task_data {task_data}')
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    def init_curious_logger(self):
+        self.logger = logging.getLogger('curious')
+        self.logger.setLevel(logging.INFO)
+        fh = logging.FileHandler(f"curious_explore_{'_'.join([str(x) for x in self.goal])}.log", 'w')
+        fh.setLevel(logging.INFO)
+        ch = logging.StreamHandler()
+        ch.setLevel(logging.INFO)
+        formatter = logging.Formatter('%(filename)s:%(lineno)s - %(funcName)s(): %(message)s')
+        fh.setFormatter(formatter)
+        self.logger.addHandler(fh)
+        self.logger.addHandler(ch)
+        self.logger.info(f'CuriousExplore task_data {self.task_data}')
+
+    @Task.step_wrapper
+    def step(self):
+        super().step()
+        self.interrupted = False
+        self.finished = False
+        
+        if self.steps[0] == "not_started":
+            if self.agent.mover.nav.is_done_exploring().value:
+                # clear memory
```

Comment:
can we have a separate piece of task_data that controls this behavior?  why should an explore always clear memory at its init?

Reply:
Moved the clearing memory out of the task to happen before the task is initialized (https://github.com/facebookresearch/fairo/pull/900/files#diff-4ca1ae20f4650f0a47b5e77eec5f8ae8c58be252584584165d5d82bbb3cbe49bR56). Note that this is in a later commit in the stack so you won't see the change here.